### ISSUE TITLE
Porblem with parsing multiple or, and logical operators

### ISSUE THREAD

**Issue by svitrol** [cortex-js/compute-engine#126]
## Description
Right so I wanted to chain multiple Or and And operators it works when I swith them but shows error when using only one kind.
(a<b) && (a<5) && (b<10) not okay
(a<b) && (a<5) || (b<10) okay

### Steps to Reproduce

1. https://cortexjs.io/mathlive/demo/
2. insert \left(0<b\right)\land\left(a<b\right)\land\left(a<-1\right)
3. 
![obrazek](https://github.com/cortex-js/compute-engine/assets/57999718/29503b7b-a68a-45dd-a04f-31b383510e62)
4. insert \left(0<b\right)\land\left(a<b\right)\lor\left(a<-1\right)
5. 
![obrazek](https://github.com/cortex-js/compute-engine/assets/57999718/d5f339e4-404f-4add-a2a7-a95210f47bf4)
6. insert \left(0<b\right)\lor\left(a<b\right)\lor\left(a<-1\right)
7. 
![obrazek](https://github.com/cortex-js/compute-engine/assets/57999718/283036d4-2890-4f75-b46c-cd2884077c2a)
8. Try to solve it with parenthesis does not help insert \left(0<b\right)\lor\left(\left(a<b\right)\lor\left(a<-1\right)\right) 
9. 
![obrazek](https://github.com/cortex-js/compute-engine/assets/57999718/4151b235-d785-4a1d-afa4-a71600dc88ea)

### Actual Behavior

From the steps above you can see when parsing multiple And and Or operators I get 'unexpcted-argument'.

### Expected Behavior

Chaining multiple And statements and Or statements should be possible.

### Environment

**MathLive version**  _If using the cortexjs.io site, the version is displayed
at the bottom of the page. If using the library, the version is available as 
`MathfieldElement.version`_
From the bottom of the site.
MathLive version
0.95.5

Updated
September 22, 2023

**Operating System** _macOS, Windows, iOS. Include the version_
OS Win11

**Browser** _Safari, Chrome, Edge, Firefox, etc... Try several browsers and note
if there are differences between browsers._
Tried several browsers does not have an impact on the result.

**Comment by svitrol** [comment]
Right so I went in to the code a bit and found some things.
https://github.com/cortex-js/compute-engine/blame/0851acb75817f4318d430f7d2b828acb01fd0899/src/compute-engine/compute-engine.ts#L265


    Functions: null, // (Anything^n) -> Anything
    NumericFunctions: null, // (Numbers^n) -> Numbers
    RealFunctions: null, // (ExtendedRealNumbers^n) -> ExtendRealNumbers
    LogicOperators: null, // (Booleans, Booleans) -> Boolean
    Predicates: null, // (Anything^n) -> Booleans

    LogicOperators: null, // (Booleans, Booleans) -> Boolean

The definition in the comment is strange. Why logic operators have only two parameters and not varying number of them?

Went a little deeper:
https://github.com/cortex-js/compute-engine/blob/0851acb75817f4318d430f7d2b828acb01fd0899/src/compute-engine/library/domains.ts#L34

  Functions: ['FunctionOf', ['VarArg', 'Anything'], 'Anything'],
  NumericFunctions: ['FunctionOf', 'Numbers', ['VarArg', 'Numbers'], 'Numbers'],
  RealFunctions: [
    'FunctionOf',
    'ExtendedRealNumbers',
    ['VarArg', 'ExtendedRealNumbers'],
    'ExtendedRealNumbers',
  ],
  LogicOperators: [
    'FunctionOf',
    'Booleans',
    ['OptArg', 'Booleans'],
    'Booleans',
  ],

Loking in this file the only domain that uses 'OptArg' is LogicOperators. 
What is the difference between 'OptArg' and 'VarArg'? From the name I would assume that varArg is any number of arguments and OptArg is max number of arguments is specified, but the definition does not say so.

From guide-domain: A value of <domain> or Nothing.

$$ \bot\land\bot\land\bot $$
Tried this:
![obrazek](https://github.com/cortex-js/compute-engine/assets/57999718/b001be67-5502-4c67-9545-aa1bdbe7aa28)
Should be parsed as ['And', False, False, False] so all three params are from the same domain. But I guess the bug is somewhere else.

After I went looking where I get 'unexpcted-argument' error. 
Getting the error here:
https://github.com/cortex-js/compute-engine/blob/0851acb75817f4318d430f7d2b828acb01fd0899/src/compute-engine/boxed-expression/validate.ts#L362

somehow here few lines above 
https://github.com/cortex-js/compute-engine/blob/0851acb75817f4318d430f7d2b828acb01fd0899/src/compute-engine/boxed-expression/validate.ts#L272
it did not get all the OptArg and did not iterate over them.


So I went in to my code and did this for And, Or operators

```
const scopeMap: RuntimeIdentifierDefinitions = this.computeEngine.context.parentScope.ids;
        if (scopeMap != null) {
            for (const operator of ['Or', 'And']) {
                const operatorDefinition = scopeMap.get(operator);
                if (operatorDefinition != null && 'signature' in operatorDefinition) {
                    const signature = operatorDefinition.signature;
                    signature.optParams = [];
                    signature.restParam = signature.params[0];
                }
            }
        }
```

Just have to set optParams to empty array and restParam set to the same value as the first param which is Boolean.
@arnog 
Was there any reason for the OptArg in logic-operators definition?

**Comment by arnog** [comment]
OK, that's some great detective work. You basically got it right. There's some info about `OptArg`/`VarArg` here https://cortexjs.io/compute-engine/guides/domains/#constructing-new-domains

Basically, `OptArg` indicates an optional argument (1 or zero), while `VarArg` indicate an argument that can be repeated zero or more times.

The issue is indeed the definition of `LogicOperators`. It should use `VarArg`, not `OptArg`.

I would also recommend avoiding modifying the "built-in" definitions, but you can override them by providing a new definition that will "shadow" them in the current scope.

**Comment by arnog** [comment]
Fixed in aa49d8b
